python/pyarrow/tests/test_table.py

```diff
@@ -878,6 +879,87 @@ def test_recordbatch_from_struct_array():
     ))
 
 
+def test_recordbatch_to_struct_array():
+    batch = pa.RecordBatch.from_arrays(
+        [
+            pa.array([1, None], type=pa.int32()),
+            pa.array([None, 1.0], type=pa.float32()),
+        ], ["ints", "floats"]
+    )
+    result = batch.to_struct_array()
+    assert result.equals(pa.array(
+        [{"ints": 1}, {"floats": 1.0}],
+        type=pa.struct([("ints", pa.int32()), ("floats", pa.float32())]),
+    ))
+
+
+def test_table_from_struct_array_invalid():
+    with pytest.raises(
+        TypeError,
+        match=re.escape(
+            "Argument 'struct_array' has incorrect type "
+            "(expected pyarrow.lib.StructArray, got pyarrow.lib.Int64Array)",
+        ),
+    ):
```

Comment:
```suggestion
    with pytest.raises(TypeError, match="Argument 'struct_array' has incorrect type"):
```

This is a bit simpler, and more than good enough I think (the actual content of the error message is generated by cython)